src/langauge-servers/configStore.ts

```diff
@@ -1,97 +1,361 @@
+                if (newIsEnabled === this.enableTypescriptServer) {
+                    return;
+                }
+                if (newIsEnabled) {
+                    this.ensureLangServerNpmFolderExists();
+                    this.installTypescriptLanguageServerIfNeeded();
+                }
+                this.enableTypescriptServer = newIsEnabled;
+                const isEnabledString = newIsEnabled ? "true" : "false";
+                this.store.set(languageStoreKey, isEnabledString);
+            } else if (language === "python") {
+                if (isEnabled === this.enablePythonServer) {
+                    return;
+                }
+                if (isEnabled) {
+                    if (!this.pythonLocation) {
+                        this.findPythonLocation();
+                    }
+                    this.installPythonLanguageServerIfNeeded();
+                }
+                this.enablePythonServer = isEnabled;
+                const isEnabledString = isEnabled ? "true" : "false";
+                this.store.set(languageStoreKey, isEnabledString);
+            } else if (language === "go") {
+                const newIsEnabled = isEnabled && isMacOrLinux;
+                if (newIsEnabled === this.enableGoServer) {
+                    return;
+                }
+                if (newIsEnabled && !this.goLocation) {
+                    this.findGoLocation();
+                }
+                this.enableGoServer = newIsEnabled;
+                const isEnabledString = newIsEnabled ? "true" : "false";
+                this.store.set(languageStoreKey, isEnabledString);
+            }
+        }
+    }
+
+    public setPythonLocation = (location: string) => {
+        if (_.isEmpty(location) || _.isEmpty(location.trim())) {
+            this.pythonLocation = "";
+            this.store.set("python-location", "");
+            return;
+        }
+        const pythonVersion = getPythonVersion(location);
+        if (pythonVersion.major >= minimumPythonMajorVersion && pythonVersion.minor >= minimumPythonMinorVersion) {
+            this.pythonLocation = location;
+            this.store.set("python-location", this.pythonLocation);
+            return;
+        } else if (pythonVersion) {
+            const errorMsg = `The submitted Python version is lower than: ${minimumPythonMajorVersion}.${minimumPythonMinorVersion}`;
+            throw new Error(errorMsg);
+        } else {
+            throw new Error("This location is an invalid Python executable location");
+        }
+    }
+
+    public setGoLocation = (location: string) => {
+        if (_.isEmpty(location) || _.isEmpty(location.trim())) {
+            this.goLocation = "";
+            this.store.set("go-location", "");
+            return;
+        }
+        const goVersion = getGoVersion(location);
+        if (goVersion.major >= minimumGoMajorVersion && goVersion.minor >= minimumGoMinorVersion) {
```

Comment:
Use a package like [this](https://www.npmjs.com/package/compare-versions) to compare version, after setting them to a string, of course